Merge develop into main

yogh333 opened the pull request.

jonored commented:
This is currently our RC1 for 2.2.4; barring changes due to feedback we will tag and promote this commit to be the 2.2.4 release.

ajinkyasraj committed:
last_amount and last_fee can be converted to total_amount and total_fee to be dual used for swap and summary.
No tests failed and amounts are accurate for both swap and batch transaction.

Co-authored-by: Sébastien Palmer <sebastien.palmer@functori.com>

ajinkyasraj committed:
Add summary page for batch transactions. Show total number of transactions, total amount and total fee.

Co-authored-by: Sébastien Palmer <sebastien.palmer@functori.com>

ajinkyasraj committed:
Change menu and settings code according to app_boilerplate from Ledger for bagl.
It simplifies maintainance.

spalmer25 committed:
[sign] extract flow functions

spalmer25 committed:
[tests] fix generated tests

ajinkyasraj committed:
Merge pull request #271 from trilitech/268-create-a-max_signing_pages-settings

Make nano home page according to Ledger app boilerplate

spalmer25 committed:
[test] rename Blind_switch screen to Accept_risk

spalmer25 committed:
[test] add tests to check the too long operation flow

 - a basic too long operation with many different operations
 - a too long operation with only transactions
 - a too long transaction without fee or amount

spalmer25 committed:
[test] add test to check too long operation flow on parsing error

spalmer25 committed:
[test] add test to check rejection of too long operations

spalmer25 committed:
[sign] add a flow to display operation summary

 - Summary of information provided later

spalmer25 committed:
[sign] add a flow for too long operation to ask to show summary

spalmer25 committed:
[sign] change Summary signing if Blindsigning is On

spalmer25 committed:
[tests] turn off blindsigning for tests with long

ajinkyasraj committed:
Add blindsigning in Settings for nano. (#275)

Options are "For Large Tx", "ON", "OFF"
Update snapshots for nano

spalmer25 committed:
[test] add tests to check blindsigning flow

spalmer25 committed:
[tests] fix touch snapshots (#279)

spalmer25 committed:
[parser] register if a micheline is the unit value

spalmer25 committed:
[tests] add tests for the staking operation

Operations:
 - stake
 - unstake
 - finalize_unstake
 - set_delegate_parameters
Check:
 - display
 - complexity

spalmer25 committed:
[test] regroup transaction tests

ajinkyasraj committed:
Add blindsigning settings ux in touch (stax/flex) (#278)

spalmer25 committed:
[memory] sort structure contents so it takes up less memory space

Also reduce `BUFF_LEN` to its smallest possible size so that the tests
continue to pass

spalmer25 committed:
Merge pull request #281 from trilitech/palmer@functori@optimize-struct

Reduce the global struct size

ajinkyasraj committed:
Merge pull request #280 from trilitech/palmer@functori@no-expert-staking-ops

Remove expert mode for `stake`/`unstake`/`finalize_unstake`

ajinkyasraj committed:
Merge pull request #276 from trilitech/palmer@functori@limit-number-of-screens-displayed-for-review

Limit the number of screens displayed for signing review for Nano

spalmer25 committed:
[sign] display expert warning only if it has not been displayed yet

spalmer25 committed:
[tests] fix the snapshots of existing tests

spalmer25 committed:
[tests] fix generated tests

spalmer25 committed:
[tests] add test to check that expert screen is displayed once

spalmer25 committed:
Merge pull request #284 from trilitech/palmer@functori@expert-mode-warning-once

Maximum one expert-mode warning per signing flow

spalmer25 committed:
[summary] finish parsing before displaying too long warning screen

spalmer25 committed:
Merge pull request #289 from trilitech/paler@functori@remove-too-long-operation-warning-on-parsing-error

[summary] remove too long operation warning on parsing error

ajinkyasraj committed:
Implement blindsigning modes on stax/flex (#286)

* Simplify blindsigning code and Use nbgl_useCaseReviewStreamingBlindSigningStart for stax

* Extract common ui_stream functions in ui_stream_common for clarity
Remove redundant code

* Add summary signing flow.
It should show warning screens and summary pages

spalmer25 committed:
[blindsign] Ask to blindsign at review in nano (#292)

* [blind] add blindsign question to jump to blindsign with blind ON

Otherwise, use the clearsign flow as the option was Large Tx

* [too_many_screen] reduce number of screen for NanoS+ and NanoX to 12

ajinkyasraj committed:
Update info page (#293)

* Add contact information in info

* Swith to large tx mode in BLINDSIGN ON mode if user does not press skip

spalmer25 committed:
[blindsign] Show blindsign warning on nano when in "Blindsign on" mode

* [sign-warning] uniform warning between nanos, nanos+ and nanox

* [blind] add warnings before going to blindsign flow

ajinkyasraj committed:
bump wallet app version (#295)

spalmer25 committed:
[blind] make blind ON acts like LARGE_TX

spalmer25 committed:
[blind] remove LARGE_TX option for nano devices

 - OFF is the default value

spalmer25 committed:
[test] remove blind tests

Redundant with test_sign_too_long_operation tests

spalmer25 committed:
[blind] remove LARGE_TX option for touch devices

 - OFF is the default value

Co-authored-by: Ajinkya <145996984+ajinkyaraj-23@users.noreply.github.com>

spalmer25 committed:
[blind] remove the too many screen flow

 - no more limit of screens (even in Blind mode ON)
 - no more warning screens for too many screen
 - large tx option acts like Blind mode OFF

Co-authored-by: Ajinkya <145996984+ajinkyaraj-23@users.noreply.github.com>

spalmer25 committed:
[blind] use boolean for blindsigning mode instead of enum

 - change blindsining mode settings from ChoiceList to Switch for stax

Co-authored-by: Ajinkya <145996984+ajinkyaraj-23@users.noreply.github.com>

spalmer25 committed:
[tests] fix snapshots according to the ledger's skip flow

ajinkyasraj committed:
Merge pull request #296 from trilitech/stax-remove-bs-large-tx

Stax-remove-bs-large-tx

ajinkyasraj committed:
Merge branch 'develop' into main

ajinkyasraj committed:
Fix static analysis warning for unreachable BAGL code for NBGL (#298)

spalmer25 committed:
Audit feedback fixes (v3.0.5) (#299)

* [format] add missing header guard to format.h file

 - plus add missing documentation

* [sign] fix assertion for message type prints

yogh333 committed:
Merge pull request #9 from trilitech/main

Add blindsigning and summary display

spalmer25 committed:
[tests] add python linter (#300)

* [tests] add python linter

* [pylint] add pylint dev routine

 - pre-commit
 - CI

* Bump ledger-app-dev-tools version to 3.42.0

* [test] fix pytezos version for python 3.9

 - fix pytezos version to the latest version compatible with python 3.9

spalmer25 committed:
[fix-touch] stop receiving data if the stream is closed (#305)

+ add testing signing micheline expression

spalmer25 committed:
Bump to 3.0.6

spalmer25 committed:
Merge pull request #307 from trilitech/palmer@functori@bump-to-3.0.6

Bump to 3.0.6

spalmer25 committed:
[test] explicits messages  (#301)

* [test] abstract Message and instantiate RawMessage

* [test] build transaction

* [test] build failing-noop

* [test] build reveal

* [test] build origination

* [test] build delegation

* [test] build register-global-constant

* [test] build transfer-ticket

* [test] build smart-rollup-add-messages

* [test] build smart-rollup-execute-outbox-message

* [test] build proposals

* [test] build ballot

* [test] build set-deposit-limit

* [test] build increase-paid-storage

* [test] build update-consensus-key

* [test] build smart-rollup-originate

* [test] build operations group

* [test] build micheline expression

ajinkyasraj committed:
Merge branch 'develop' into main

Leereen committed:
Merge pull request #10 from trilitech/main

Fix flex/stax bug for signing Michelin messages

mruiz-ledger committed:
fix: variables no longer exist

mruiz-ledger committed:
Merge pull request #11 from LedgerHQ/pr-io-revamp

fix: variables no longer exist

mruiz-ledger committed:
Bump version to 3.0.7

mruiz-ledger committed:
Merge pull request #12 from LedgerHQ/mru/bump-version

Bump version
